src/common/RAJAPerfSuite.hpp

```diff
@@ -233,6 +233,63 @@ enum FeatureID {
 };
 
 
+/*!
+ *******************************************************************************
+ *
+ * \brief Enumeration defining unique id for each (CUDA) Data memory space
+ * used in suite.
+ *
+ * IMPORTANT: This is only modified when a new memory space is used in suite.
+ *
+ *            IT MUST BE KEPT CONSISTENT (CORRESPONDING ONE-TO-ONE) WITH
+ *            ARRAY OF MEMORY SPACE NAMES IN IMPLEMENTATION FILE!!!
+ *
+ *******************************************************************************
+ */
+enum struct CudaDataSpace {
+
+  Host = 0,
+  Pinned,
+  Managed,
+  Device,
+
+  NumSpaces // Keep this one last and NEVER comment out (!!)
+
+};
+
+
+/*!
+ *******************************************************************************
+ *
+ * \brief Enumeration defining unique id for each (HIP) Data memory space
+ * used in suite.
+ *
+ * IMPORTANT: This is only modified when a new memory space is used in suite.
+ *
+ *            IT MUST BE KEPT CONSISTENT (CORRESPONDING ONE-TO-ONE) WITH
+ *            ARRAY OF MEMORY SPACE NAMES IN IMPLEMENTATION FILE!!!
+ *
+ *******************************************************************************
+ */
+enum struct HipDataSpace {
+
+  Host = 0,
+  HostAdviseFine,
+  HostAdviseCoarse,
+  Pinned,
+  PinnedFine,
+  PinnedCoarse,
+  Managed,
+  ManagedAdviseFine,
+  ManagedAdviseCoarse,
+  Device,
+  DeviceFine,
+
+  NumSpaces // Keep this one last and NEVER comment out (!!)
```

Comment:
Is there a reason we have NumSpaces in the enum and just not a constexpr member variable?